services/QuillLMS/app/controllers/cms/districts_controller.rb

```diff
@@ -0,0 +1,253 @@
+# frozen_string_literal: true
+
+class Cms::DistrictsController < Cms::CmsController
+  before_action :signed_in!
+
+  before_action :text_search_inputs, only: [:index, :search]
+  before_action :set_district, only: [
+    :show
+  ]
+
+  DISTRICTS_PER_PAGE = 30.0
+
+  def index
+    @district_search_query = {}
+    @district_search_query_results = []
+    @number_of_pages = 0
+  end
+
+  def search
+    district_search_query = district_query_params
+    district_search_query_results = district_query(district_query_params)
+    district_search_query_results ||= []
+    number_of_pages = (number_of_districts_matched / DISTRICTS_PER_PAGE).ceil
+    render json: {numberOfPages: number_of_pages, districtSearchQueryResults: district_search_query_results}
+  end
+
+  def show
+    @district = {
+      'Name' => @district.name,
+      'City' => @district.city,
+      'State' => @district.state,
+      'ZIP' => @district.zipcode,
+      'Clever ID' => @district.clever_id,
+      'NCES ID' => @district.nces_id,
+      'Phone' => @district.phone,
+      'Total Students' => @district.total_students,
+      'Total Schools' => @district.total_schools,
+      'Grade Range' => @district.grade_range,
+      'Total Invoice Amount' => @district.total_invoice
+    }
+    @school_data = school_query
+    @admins = DistrictsAdmins.includes(:user).where(district_id: params[:id].to_i).map do |admin|
+      {
+        name: admin.user.name,
+        email: admin.user.email,
+        district_id: admin.district_id,
+        user_id: admin.user_id
+      }
+    end
+  end
+
+  def new_admin
+    @district = District.find(params[:id])
+  end
+
+  def add_admin_by_email
+    begin
+      user = User.find_by(email: params[:email_address])
+      district = District.find(params[:id])
+      DistrictsAdmins.create(user_id: user.id, district_id: district.id)
+      flash[:success] = "Yay! It worked! 🎉"
+      redirect_to cms_district_path(params[:id])
+    rescue
```

Comment:
This will only rescue if the district isn't found. Use `create!` if you'd like it to `raise`